pkg/cli/clisqlclient/conn.go

```diff
@@ -191,17 +193,39 @@ func (c *sqlConn) EnsureConn() error {
 	return nil
 }
 
+type showLastQueryStatsMode int
+
+const (
+	modeDisabled showLastQueryStatsMode = iota
+	modeModern
+	modeSimple // Remove this when pre-21.2 compatibility is not needed any more.
+)
+
 // tryEnableServerExecutionTimings attempts to check if the server supports the
 // SHOW LAST QUERY STATISTICS statements. This allows the CLI client to report
 // server side execution timings instead of timing on the client.
 func (c *sqlConn) tryEnableServerExecutionTimings() {
-	_, _, _, _, _, _, err := c.getLastQueryStatisticsInternal()
-	if err != nil {
-		fmt.Fprintf(c.errw, "warning: cannot show server execution timings: %v\n", err)
-		c.connCtx.EnableServerExecutionTimings = false
-	} else {
+	// Starting in v21.2 servers, clients can request an explicit set of
+	// values which makes them compatible with any post-21.2 column
+	// additions.
+	_, err := c.QueryRow("SHOW LAST QUERY STATISTICS RETURNING x", nil)
+	if err == nil {
```

Comment:
should we catch for syntax errors / do retries?
a transient error may disable query statistics for the entire session. maybe we want to log that.